src/libraries/System.Private.CoreLib/src/System/ThrowHelper.cs

```diff
@@ -1244,6 +1250,12 @@ private static string GetResourceString(ExceptionResource resource)
                     return SR.Format_ExpectedAsciiDigit;
                 case ExceptionResource.Argument_HasToBeArrayClass:
                     return SR.Argument_HasToBeArrayClass;
+                case ExceptionResource.Arg_PrimitiveWiden:
```

Comment:
I do not think these exceptions needs to be factored into `ThrowHelper`. They are not used in perf sensisitive places.

Reply:
Doesn't InitializeArray affect startup performance?

Reply:
It is not a hot method. The ThrowHelper makes sense in hot (typically very small) methods; or in generics where the code is stamped many times.